cocos/scripting/js-bindings/manual/jsb_cocos2dx_manual.cpp

```diff
@@ -836,6 +819,89 @@ static bool js_EventListenerCustom_create(se::State& s)
 }
 SE_BIND_FUNC(js_EventListenerCustom_create)
 
+static void onBeforeDispatchTouchEvent(Event* event)
+{
+    if (event->getType() == Event::Type::TOUCH)
+    {
+        EventTouch* touchEvent = static_cast<EventTouch*>(event);
+        if (touchEvent->getEventCode() == EventTouch::EventCode::BEGAN)
+        {
+            bool fromCache = true;
+            const auto& touches = touchEvent->getTouches();
+            for (auto&& touch : touches)
+            {
+                se::Value touchVal;
+                native_ptr_to_seval<Touch>(touch, __jsb_cocos2d_Touch_class, &touchVal, &fromCache);
+                assert(!fromCache);
+                touchVal.toObject()->root();
```

Comment:
Cocos2D-X reuses `Touch` pointer from touch began to touch ended(cancelled) when dispatching a touch event. We only need to root it from touch began and unroot it in touch ended(cancelled).